### PR TITLE
Fix Attendee element

### PR DESCRIPTION
**Pull request by SebastianBecker2**
The Attendee element was missing potential ProposedStart and
ProposedEnd child elements. Those were missing from Microsoft
documentation. They are used when an attendee proposes a different
date and time for the meeting.

I already created a pull request for Microsofts docs. But the types
schema shows that ProposedStart and ProposedEnd are possible
child elements.
BTW: The Mailbox child element is now mandatory. I'm pretty sure
this is correct. But the type schema doesn't sepecify this element with
'required'.

### CODE REVIEW

File: include/ews/ews.hpp [hunk 1]
```diff
@@ -12649,6 +12676,16 @@ class attendee final
              << "</t:ResponseType>";
         sstr << "<t:LastResponseTime>" << last_response_time_.to_string()
              << "</t:LastResponseTime>";
+        if (proposed_start_.to_string().empty())
+        {
+            sstr << "<t:ProposedStart>" << proposed_start_.to_string()
+                 << "</t:ProposedStart>";
```
Comment (idolum) [inline include/ews/ews.hpp]: Do I see this right that the proposed start is only added to the XML, if it's empty? I would expect it the other way around.

File: include/ews/ews.hpp [hunk 2]
```diff
@@ -12649,6 +12676,16 @@ class attendee final
...
+        }
+        if (proposed_end_.to_string().empty())
+        {
+            sstr << "<t:ProposedEnd>" << proposed_end_.to_string()
+                 << "</t:ProposedEnd>";
```
Comment (idolum) [inline include/ews/ews.hpp]: The same as above.

File: include/ews/ews.hpp [hunk 3]
```diff
@@ -12678,6 +12717,17 @@ class attendee final
         create_node(attendee_node, "t:LastResponseTime",
                     last_response_time_.to_string());
 
+        if (proposed_start_.to_string().empty())
+        {
+            create_node(attendee_node, "t:ProposedStart",
+                        proposed_start_.to_string());
```
Comment (idolum) [inline include/ews/ews.hpp]: The same as above.

File: include/ews/ews.hpp [hunk 4]
```diff
@@ -12678,6 +12717,17 @@ class attendee final
...
+        }
+        if (proposed_end_.to_string().empty())
+        {
+            create_node(attendee_node, "t:ProposedEnd",
+                        proposed_end_.to_string());
```
Comment (idolum) [inline include/ews/ews.hpp]: The same as above.

File: include/ews/ews.hpp [hunk 5]
```diff
@@ -12690,44 +12740,52 @@ class attendee final
...
         //    <ResponseType/>
         //    <LastResponseTime/>
+        //    <ProposedStart/>
+        //    <ProposedEnd/>
         //  </Attendee>
 
-        mailbox addr;
-        auto resp_type = response_type::unknown;
-        date_time last_resp_time("");
+        auto mailbox_node = elem.first_node("t:Mailbox");
```
Comment (idolum) [inline include/ews/ews.hpp]: I wonder, if `first_node_ns(internal::uri<>::microsoft::types(), "Maibox")` would be more appropriate, as it does not rely on the namespace prefix. Might be different in a received XML message.

**Comment by SebastianBecker2** [comment]
😢
I made a mistake...
I'm trying to fix it...

File: include/ews/ews.hpp (hunk 1 above)
Comment (SebastianBecker2) [reply include/ews/ews.hpp]: Fixed it. Thx.

File: include/ews/ews.hpp (hunk 5 above)
Comment (SebastianBecker2) [reply include/ews/ews.hpp]: Fixed it. Thx.

**Comment by SebastianBecker2** [comment]
I will never try to amend a commit that is already in a remote which is in the process of a PR.
But I think I fixed it.

**Comment by idolum** [comment]
Looks good! Thanks!

**Pull request merged**